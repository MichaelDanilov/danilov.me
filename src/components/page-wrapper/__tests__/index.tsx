import React from 'react';
import renderer from 'react-test-renderer';
import PageWrapper from '..';

describe('PageWrapper', () => {
  it('english version renders correctly', () => {
    const tree = renderer
      .create(
        <PageWrapper lang="en" meta={{}}>
          <span>Lorem ipsum</span>
        </PageWrapper>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('russian version renders correctly', () => {
    const tree = renderer
      .create(
        <PageWrapper lang="ru" meta={{}}>
          <span>Lorem ipsum</span>
        </PageWrapper>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
