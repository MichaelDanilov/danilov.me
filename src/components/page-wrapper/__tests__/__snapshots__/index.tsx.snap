// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageWrapper english version renders correctly 1`] = `
Array [
  <div
    className="sc-ifAKCX kjRRvg"
  >
    <h1
      className="sc-bdVaJa gyVQum"
    >
      Michael Danilov
    </h1>
    <ul
      className="sc-htpNat datbWf"
    >
      <li
        className="sc-bxivhb jwZqTZ"
      >
        <a
          aria-current="page"
          className=""
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
          style={Object {}}
        >
          Home
        </a>
      </li>
      <li
        className="sc-bxivhb jwZqTZ"
      >
        <a
          href="/contacts"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Contacts
        </a>
      </li>
      <li
        className="sc-bxivhb jwZqTZ"
      >
        <a
          href="/ru"
          hrefLang="ru-RU"
          lang="ru-RU"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Русская версия
        </a>
      </li>
    </ul>
    <span>
      Lorem ipsum
    </span>
  </div>,
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<!-- Yandex.Metrika counter --> <script type=\\"text/javascript\\" > (function(m,e,t,r,i,k,a){m[i]=m[i]||function(){(m[i].a=m[i].a||[]).push(arguments)}; m[i].l=1*new Date();k=e.createElement(t),a=e.getElementsByTagName(t)[0],k.async=1,k.src=r,a.parentNode.insertBefore(k,a)}) (window, document, \\"script\\", \\"https://mc.yandex.ru/metrika/tag.js\\", \\"ym\\"); ym(990590, \\"init\\", { id:990590, clickmap:true, trackLinks:true, accurateTrackBounce:true, webvisor:true }); </script> <noscript><div><img src=\\"https://mc.yandex.ru/watch/990590\\" style=\\"position:absolute; left:-9999px;\\" alt=\\"\\" /></div></noscript> <!-- /Yandex.Metrika counter -->",
      }
    }
  />,
]
`;

exports[`PageWrapper russian version renders correctly 1`] = `
Array [
  <div
    className="sc-ifAKCX kjRRvg"
  >
    <h1
      className="sc-bdVaJa gyVQum"
    >
      Михаил Данилов
    </h1>
    <ul
      className="sc-htpNat datbWf"
    >
      <li
        className="sc-bxivhb jwZqTZ"
      >
        <a
          href="/ru"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Главная
        </a>
      </li>
      <li
        className="sc-bxivhb jwZqTZ"
      >
        <a
          href="/ru/contacts"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Контакты
        </a>
      </li>
      <li
        className="sc-bxivhb jwZqTZ"
      >
        <a
          aria-current="page"
          className=""
          href="/"
          hrefLang="en-US"
          lang="en-US"
          onClick={[Function]}
          onMouseEnter={[Function]}
          style={Object {}}
        >
          English version
        </a>
      </li>
    </ul>
    <span>
      Lorem ipsum
    </span>
  </div>,
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<!-- Yandex.Metrika counter --> <script type=\\"text/javascript\\" > (function(m,e,t,r,i,k,a){m[i]=m[i]||function(){(m[i].a=m[i].a||[]).push(arguments)}; m[i].l=1*new Date();k=e.createElement(t),a=e.getElementsByTagName(t)[0],k.async=1,k.src=r,a.parentNode.insertBefore(k,a)}) (window, document, \\"script\\", \\"https://mc.yandex.ru/metrika/tag.js\\", \\"ym\\"); ym(990590, \\"init\\", { id:990590, clickmap:true, trackLinks:true, accurateTrackBounce:true, webvisor:true }); </script> <noscript><div><img src=\\"https://mc.yandex.ru/watch/990590\\" style=\\"position:absolute; left:-9999px;\\" alt=\\"\\" /></div></noscript> <!-- /Yandex.Metrika counter -->",
      }
    }
  />,
]
`;
